### ISSUE TITLE
Reduce test data size? 

### ISSUE THREAD

**Issue by andybega** [ShanaScogin/BayesPostEst#79]
The test data is the main contributor to overall package size, pushing it close/over the 5 MB limit that generates R check NOTE. Is is possible to cut down the size of some of these object, or eliminate them?

From https://github.com/ShanaScogin/BayesPostEst/issues/77:

> FWIW, looking at the built package tarball, seems that the BRMS and RStan logit examples in `tests/testdata` are the big size offenders. Maybe it's possible to cut some elements from those objects that are not really needed in the tests. (Like how you can reduce the size of a fitted GLM model by cutting out the training data copy, etc.)
> 
> (`tests/testdata` itself is 4.1 MB)
>
> ![Screen Shot 2021-01-06 at 10 50 07](https://user-images.githubusercontent.com/1353756/103748530-f24e9d00-500c-11eb-96cd-f1d971687487.png)

**Comment by jayrobwilliams** [comment]
I think it should be possible to save fewer samples on those objects at the very least. I'll look at that today.

**Comment by jayrobwilliams** [comment]
I've cut the size of `rstan_logit` approximately in half by not saving warmup iterations. I've done the same and gotten `brms_logit` down to 288.8 kb in R, but when I save either of them, the resulting `.rds` file barely shrinks. I'm going to poke around a bit more and see if I can figure out what's going on here.

This is known behavior depending on the sparsity/complexity of vectors and matrices to be saved https://stackoverflow.com/questions/24539106/what-determines-the-size-of-a-saved-object-in-r

**Comment by jayrobwilliams** [comment]
Managed to shrink them a bit.
<img width="564" alt="Screen Shot 2021-01-06 at 1 32 56 PM" src="https://user-images.githubusercontent.com/12078284/103807010-cd195900-5023-11eb-89db-4d33717981f8.png">

This passes `devtools::check()` on my machine with no issues. @ShanaScogin can you check it with winbuilder, etc.? Changes are in https://github.com/ShanaScogin/BayesPostEst/commit/dfed862bbbb8be8c1d8f95f836a5aa7c24fb4ccb

**Comment by ShanaScogin** [comment]
Ya! I'll do! If yall want to try with the ones in the main folder, that'd probably also be good. I deleted the bugs ones, the interactive ones, and the mcmcpack ones, but the code is still in the data-raw to run and re do the files.

![image](https://user-images.githubusercontent.com/26779067/103807599-a0b20c80-5024-11eb-959a-3d921512a3d2.png)

**Comment by jayrobwilliams** [comment]
It also looks like `jags_interactive_cat`, `jags_interactive` and `jags_probit` are only used in tests; should I move them to the test data folder? At this point, `jags_logit` is the only one used in function documentation examples.

**Comment by ShanaScogin** [comment]
@jayrobwilliams Ya great idea! Since the test files still make it into the CRAN build, it won't help the size issue, but I think it makes sense for organization, personally. Up to you.

**Comment by ShanaScogin** [comment]
I picked this up again since I wanted to push an unrelated fix to CRAN. I have attempted to complete this in commit ebbfc34. I added a file called setup.R that runs the .R files in the setup-data file (previously in 'testdata-raw') and then cleans them up after tests are run. I've saved the files in data-static that is ignored by the build just for our own purposes, but the tests are designed to work with the temp ones. Right now, the package passes a local testthat test, winbuilder release, and a CMD check. Need to do rhub. There is still a question of R2jags (notes in file helper-libs.R though now attaching in setup.R, so a bit redundant), but that falls under the purview of into Issue #66. The test files and test data definitely still need a clean, but I might make a separate issue since this one is technically solved.
